src/box/txn_limbo.c

```diff
@@ -363,6 +415,11 @@ void
 txn_limbo_checkpoint(const struct txn_limbo *limbo, struct synchro_request *req,
 		     struct vclock *vclock)
 {
+	/* For simplicity we prohibit checkpoints during leadership changes. */
+	if (!rlist_empty((struct rlist *)&limbo->pending_promotes))
+		say_warn("limbo checkpoint: cannot capture pending PROMOTEs");
```

Comment:
1. That looks like a warning which won't prohibit the snapshot, no? If it is prohibited on an upper level, then shouldn't this be an assert?
2. Why do we prohibit it? What happens if the limbo is empty, we make a snapshot with a non-empty promote queue, and restart? Wouldn't the node then just re-enter the elections gracefully? I.e. bump term, start new elections, etc.